UI/ChatGPT/ChatGPT/ChatGPT/Presentation/MainModel.cs

```diff
@@ -47,18 +47,16 @@ private async ValueTask Ask(string prompt, CancellationToken ct)
 		var message = Message.CreateLoading();
 		await Messages.AddAsync(message, ct);
 
-		var history = (await Messages)
-			.Where(msg => msg.Status is Status.Value)
-			.Select(msg => new ChatEntry(msg.Content!, msg.Source is Source.User))
-			.ToImmutableList();
+		var requestContent = await CreateRequest();
 
-		//Create ChatRequest record with history (Messages list)
-		var request = new ChatRequest(history);
+		//Create ChatRequest record with messages list
+		var request = new ChatRequest(requestContent);
 
 		var response = await _chatService.AskAsync(request);
 
 		//Update loading message with AI response as Message record
-		await Update(message, response, ct);
+		//Finds the message with same id and upadtes the instance
+		await Messages.UpdateAsync(message.Update(response), ct);
```

Comment:
Does this add to the list of Messages, or amend an existing message - if the latter, feels very not-immutable :-)

Reply:
It looks for an item with same `Id` and if the item is found, the item is removed and the updated (new) item is added.

[Here is the `UpdateAsync`](https://github.com/unoplatform/uno.extensions/blob/main/src/Uno.Extensions.Reactive/Core/ListState.Extensions.cs#L199) code in Uno.Extensions.Reactive.

Reply:
ListState is "Mutable" conceptually... Just like State.... This is why it exists in the first place....

It's all good